lib/foreman/http_proxy/excon_connection_extension.rb

```diff
@@ -0,0 +1,13 @@
+Excon::Connection.class_eval do
+  include Foreman::HTTPProxy
+  alias_method :orig_request, :request
+
+  def request(params, &block)
+    if proxy_http_request?(@data[:proxy], @data[:host], @data[:schema])
+      log_proxied_request(http_proxy, @data[:host])
+      @data[:proxy] = http_proxy
+      setup_proxy
```

Comment:
The only way that comes to mind for validating the HTTP proxy setting would be to ping an arbitrary URL, which can have other reasons of failing too.

It may be more reasonable to rescue these errors and show a meaningful error message.
Agree?